src/tests/casesSlice.test.js

```diff
@@ -0,0 +1,204 @@
+import configureStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import axios from 'axios';
+import {
+  setColor, getCases, getHistoricallData,
+} from '../redux/cases/casesSlice';
+
+jest.mock('axios', () => ({
+  __esModule: true,
+  default: jest.fn(),
+}));
+
+const middlewares = [thunk];
+const mockStore = configureStore(middlewares);
+
+describe('Redux actions', () => {
+  let store;
+
+  beforeEach(() => {
+    store = mockStore({});
+  });
+
+  afterEach(() => {
+    store.clearActions();
+    jest.restoreAllMocks();
+  });
+
+  describe('getCases', () => {
+    it('dispatches the correct actions on successful API call', async () => {
+      const mockResponse = {
+        country: 'Country',
+        population: 1000000,
+        cases: 1000,
+        todayCases: 100,
+        deaths: 100,
+        todayDeaths: 10,
+        recovered: 900,
+        todayRecovered: 50,
+        active: 100,
+        critical: 20,
+        casesPerOneMillion: 1000,
+        recoveredPerOneMillion: 900,
+        deathsPerOneMillion: 100,
+      };
+
+      axios.mockResolvedValueOnce({ data: mockResponse });
+
+      const expectedActions = [
+        {
+          type: 'cases/getCases/pending',
+          payload: undefined,
+          meta: {
+            arg: 'inputcountry',
+            requestId: expect.any(String),
+            requestStatus: 'pending',
+          },
+        },
+        {
+          type: 'cases/getCases/fulfilled',
+          payload: mockResponse,
+          meta: {
+            arg: 'inputcountry',
+            requestId: expect.any(String),
+            requestStatus: 'fulfilled',
+          },
+        },
+      ];
+
+      await store.dispatch(getCases('inputcountry'));
+
+      expect(store.getActions()).toEqual(expectedActions);
+    });
+
+    it('dispatches the correct actions on API call failure', async () => {
+      const mockError = new Error('API error');
+
+      axios.mockRejectedValueOnce(mockError);
+
+      const expectedActions = [
+        {
+          type: 'cases/getCases/pending',
+          payload: undefined,
+          meta: {
+            arg: 'inputcountry',
+            requestId: expect.any(String),
+            requestStatus: 'pending',
+          },
+        },
+        {
+          type: 'cases/getCases/rejected',
+          payload: undefined,
+          meta: {
+            arg: 'inputcountry',
+            requestId: expect.any(String),
+            rejectedWithValue: false,
+            requestStatus: 'rejected',
+            aborted: false,
+            condition: false,
+          },
+          error: {
+            name: 'Error',
+            message: 'API error',
+            stack: expect.any(String),
+          },
+        },
+      ];
+      await store.dispatch(getCases('inputcountry'));
+
+      expect(store.getActions()).toEqual(expectedActions);
+    });
+  });
+
+  describe('getHistoricallData', () => {
```

Comment:
Nice job writing some test cases for your redux code👍. You are also required to write integration tests for your react components making use of the react-testing-library😉. Kindly do well to implement that👍.
<img width="611" alt="image" src="https://github.com/EmiLoKiJu/covid-data/assets/64643995/7cbe1a72-b303-4eca-8b8f-bf88ddcb3eec">